tests/test_abstract.py

```diff
@@ -28,6 +29,14 @@ def test_to_abstract():
     assert to_abstract(inst) == expected
 
 
+def test_numpy_scalar_to_abstract():
+    s1 = AbstractScalar({VALUE: 2, TYPE: i64})
+    assert to_abstract_test(np.int64(2)) == s1
```

Comment:
int64 is probably a bad test since it's de default type for python ints.  Maybe try int16?